modules/fluid_properties/unit/src/CaloricallyImperfectGasTest.C

```diff
@@ -0,0 +1,510 @@
+    Real e = 10342290 * 0.75 + 17034360 * 0.25;
+    Real h = e + p / fp->rho_from_p_T(p, T);
+
+    Real h_pT = fp->h_from_p_T(p, T);
+    Real e_pT = fp->e_from_p_T(p, T);
+    REL_TEST(h, h_pT, 1e-5);
+    REL_TEST(e, e_pT, 1e-5);
+
+    Real v = fp->v_from_p_T(p, T);
+    Real e_vh = fp->e_from_v_h(v, h);
+    REL_TEST(e, e_vh, 1e-5);
+  }
+
+  // check cv/cp lookups for T = 325.0
+  {
+    Real T = 325.0;
+    Real p = 1.0e6;
+    Real cv = (17034360.0 - 10342290.0) / 100.0;
+    Real cp = cv + 8.3144598 / 0.002;
+    Real cp_pT = fp->cp_from_p_T(p, T);
+    Real cv_pT = fp->cv_from_p_T(p, T);
+    REL_TEST(cp, cp_pT, 1e-5);
+    REL_TEST(cv, cv_pT, 1e-5);
+
+    Real v = fp->v_from_p_T(p, T);
+    Real e = fp->e_from_p_T(p, T);
+    Real cp_ve = fp->cp_from_v_e(v, e);
+    Real cv_ve = fp->cv_from_v_e(v, e);
+    REL_TEST(cp, cp_ve, 1e-5);
+    REL_TEST(cv, cv_ve, 1e-5);
+  }
+
+  // test mu(p, T), mu(v, e), k(p, T), k(v, e) & derivatives
+  {
+    Real T = 250.0;
+    Real p = 1.0e6;
+    REL_TEST(fp->mu_from_p_T(p, T), 2.875, 1e-5);
+    REL_TEST(fp->k_from_p_T(p, T), 7.25, 1e-5);
+
+    Real v = fp->v_from_p_T(p, T);
+    Real e = fp->e_from_p_T(p, T);
+    REL_TEST(fp->mu_from_v_e(v, e), 2.875, 0.001);
+    REL_TEST(fp->k_from_v_e(v, e), 7.25, 0.001);
+
+    Real mu, dmu_dp, dmu_dT;
+    fp->mu_from_p_T(p, T, mu, dmu_dp, dmu_dT);
+    REL_TEST(dmu_dT, 0.0125, 1e-5);
+
+    Real k, dk_dp, dk_dT;
+    fp->k_from_p_T(p, T, k, dk_dp, dk_dT);
+    REL_TEST(dk_dT, -0.025, 1e-5);
+
+    Real cv = fp->cv_from_p_T(p, T);
+    Real dmu_dv, dmu_de;
+    fp->mu_from_v_e(v, e, mu, dmu_dv, dmu_de);
+    REL_TEST(dmu_de, 0.0125 / cv, 0.001);
+
+    Real dk_dv, dk_de;
+    fp->k_from_v_e(v, e, k, dk_dv, dk_de);
+    REL_TEST(dk_de, -0.025 / cv, 0.001);
+  }
+
+  // test entropy functions
+  {
+    Real T = 489.305;
+    Real Z = 0.5 * (80459 + 80460.7);
+    Real p = 1.0e6;
+    const Real rho = fp->rho_from_p_T(p, T);
+    const Real v = 1.0 / rho;
+    const Real v0 = 1.0;
+    Real e = fp->e_from_p_T(p, T);
+    Real h = fp->h_from_p_T(p, T);
+    Real s = Z + Rs * std::log(v / v0);
+
+    Real s_pT = fp->s_from_p_T(p, T);
+    REL_TEST(s_pT, s, 1e-6);
+
+    Real s_ve = fp->s_from_v_e(v, e);
+    REL_TEST(s_ve, s, 1e-6);
+
+    Real s_hp = fp->s_from_h_p(h, p);
+    REL_TEST(s_hp, s, 1e-6);
+
+    Real s_Tv = fp->s_from_T_v(T, v);
+    REL_TEST(s_Tv, s, 1e-6);
+
+    // test 5 arg functions with derivatives for s(p, T)
+    {
+      Real s_alt, ds_dp, ds_dT;
+      fp->s_from_p_T(p, T, s_alt, ds_dp, ds_dT);
+      REL_TEST(s_alt, s, 1e-5);
+      Real eps = 1e-4;
+      Real pp1 = p * (1 + eps);
+      Real pp2 = p * (1 - eps);
+      Real TT1 = T * (1 + eps);
+      Real TT2 = T * (1 - eps);
+      Real ds_dp_fd = (fp->s_from_p_T(pp1, T) - fp->s_from_p_T(pp2, T)) / (2 * eps * p);
+      REL_TEST(ds_dp_fd, ds_dp, 1e-3);
+      Real ds_dT_fd = (fp->s_from_p_T(p, TT1) - fp->s_from_p_T(p, TT2)) / (2 * eps * T);
+      REL_TEST(ds_dT_fd, ds_dT, 1e-3);
```

Comment:
You can do a derivative test with `DERIV_TEST(fp->s_from_p_T, p, T, 1e-3)` (`eps = 1e-6`). Or if you need `eps = 1e-4`, `DERIV_TEST_CUSTOM_PERTURBATION(fp->s_from_p_T, p, T, 1e-3, 1e-4)`.